packages/quantic/force-app/main/default/lwc/quanticDidYouMean/quanticDidYouMean.js

```diff
@@ -67,25 +83,40 @@ export default class QuanticDidYouMean extends LightningElement {
   initialize = (engine) => {
     this.headless = getHeadlessBundle(this.engineId);
     this.didYouMean = this.headless.buildDidYouMean(engine);
-    this.unsubscribe = this.didYouMean.subscribe(() => this.updateState());
+    this.queryTrigger = this.headless?.buildQueryTrigger?.(engine);
+    this.unsubscribeDidYouMean = this.didYouMean.subscribe(() =>
+      this.updateDidYouMeanState()
+    );
+    this.unsubscribeQueryTrigger = this.queryTrigger?.subscribe(() =>
```

Comment:
Why can the `this.queryTrigger.subscribe` be null but not the `this.didYouMean.subscribe`?

Reply:
yeah not sure I understand that part

Reply:
Because the QueryTrigger is not yet available in the Insight Use Case, currently the query trigger Headless controller only in the search use case, while the Headless did you mean controller is available in both use cases.

that's why I'm adding these checks, to avoid breaking the Quantic Did You Mean component when being used in the insight use case.

We will need to update Headless in the futur tho, to make it support the triggers for the insight use case.